specification/msi/resource-manager/Microsoft.ManagedIdentity/2015-08-31-preview/ManagedIdentity.json

```diff
@@ -0,0 +1,535 @@
+      "x-ms-external": true,
+      "properties": {
+        "error": {
+          "description": "A list of additional details about the error.",
+          "$ref": "#/definitions/CloudErrorBody"
+        }
+      },
+      "description": "An error response from the ManagedServiceIdentity service."
+    },
+    "CloudErrorBody": {
+      "x-ms-external": true,
+      "properties": {
+        "code": {
+          "type": "string",
+          "description": "An identifier for the error."
+        },
+        "message": {
+          "type": "string",
+          "description": "A message describing the error, intended to be suitable for display in a user interface."
+        },
+        "target": {
+          "type": "string",
+          "description": "The target of the particular error. For example, the name of the property in error."
+        },
+        "details": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/CloudErrorBody"
+          },
+          "description": "A list of additional details about the error."
+        }
+      },
+      "description": "An error response from the ManagedServiceIdentity service."
+    },
+    "Operation": {
+      "title": "Microsoft.ManagedIdentity REST API operation",
+      "type": "object",
+      "properties": {
+        "name": {
+          "title": "The operation name.",
+          "description": "This is of the format {provider}/{resource}/{operation}",
+          "type": "string"
+        },
+        "display": {
+          "title": "The object that describes the operation.",
+          "properties": {
+            "provider": {
+              "title": "Friendly name of the resource provider.",
+              "type": "string"
+            },
+            "operation": {
+              "title": "The operation type.",
+              "description": "For example: read, write, delete",
+              "type": "string"
+            },
+            "resource": {
+              "title": "The resource type on which the operation is performed.",
+              "type": "string"
+            },
+            "description": {
+              "title": "The friendly name of the operation",
+              "type": "string"
+            }
+          }
+        }
+      }
+    },
+    "OperationListResult": {
+      "title": "Result of the request to list REST API operations. It contains a list of operations supported by the API.",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Operation"
+          },
+          "title": "The list of operations supported by the resource provider."
+        }
+      }
+    }
+  },
+  "parameters": {
+    "SubscriptionIdParameter": {
+      "name": "subscriptionId",
+      "in": "path",
+      "description": "The Id of the Subscription to which the identity belongs.",
+      "required": true,
+      "type": "string",
+      "x-ms-parameter-location": "method"
+    },
```

Comment:
Are you sure you want the location to be the method instead of the client?

Reply:
Well we get the subscriptionId, resourceGroupName etc from the resource id/path. What is the procedure followed by other Azure resources. Azure resources in general get the subscriptionId etc from the path. We should follow what they do

Reply:
I am not asking for moving subID to a place different than the path.
What I mean is, in the generated SDKs there is the option to generate this parameter as a parameter in all operation methods, or as a client property. This way the method signature will be shorter, and users wont need to pass a value (that is likely wont change very often) again and again. [More about x-ms-parameter-location](https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-parameter-location)

Reply:
Changed location of subscriptionId parameter to client